go/vt/sqlparser/token.go

```diff
@@ -651,7 +651,7 @@ func (tkn *Tokenizer) skipBlank() {
 func (tkn *Tokenizer) scanIdentifier(firstByte byte, isDbSystemVariable bool) (int, []byte) {
 	buffer := &bytes2.Buffer{}
 	buffer.WriteByte(firstByte)
-	for isLetter(tkn.lastChar) || isDigit(tkn.lastChar) || (isDbSystemVariable && isCarat(tkn.lastChar)) {
+	for isLetter(tkn.lastChar) || isDigit(tkn.lastChar) || tkn.lastChar == '$' || (isDbSystemVariable && isCarat(tkn.lastChar)) {
```

Comment:
As discussed yesterday, `$` seems to be valid at the beginning of an identifier also. So, it's better to add it to `isLetter`.

Reply:
Ping